### DISCUSSION TITLE
Sherlock v2

### DISCUSSION THREAD

**Discussion by Blade2021** [Blade2021/Sherlock#18] [Ideas]
What are some things you wish Sherlock could do for your server?

**Comment by SirGoodenough** [comment]
Grab Amazon, Bangood, and AliExpress links as they pop up and repost them with the affiliate tags in the URL for others to use.  Example Digiblur is doing it.

**Reply by Blade2021** [reply]
This doesn't really fit the job for Sherlock.

**Reply by SirGoodenough** [reply]
That's fine.  I thought it may be something that other guilds could use is all.

**Comment by SirGoodenough** [comment]
Improve language filter such that spaces can be added to the filter.
This is so that a-s-s can be filtered and some specific derivatives, but associate would not be flagged.

**Comment by SirGoodenough** [comment]
Posting this discussion here as a V2 straw-man, something to stop channel spammers...
The channel name could be numbers or it could be 'do not post here you will be muted' or something that makes it hard to mistake.  Just thinking making its obscurely named and so it shows first alpha-numerically so the spammers hit it first.  You could have Sherlock drop the .topic as a reply to any message and that explains what the channel is and if you messed up and are muted because you sent a message here, how to fix it.  The mute could be for 10 minutes or something, that's all it would take...

(https://discord.com/channels/469330414121517056/469558321590960150/882680025244114984)
Sir_GoodEnough_🛠🔶 — 09/01/2021
Have to have @Blade2021🔶 opinion if this is feesable, but since we are unlikely to have a user named spam  If Hive watched for @ spam tag and when seen, drop a message in the radiants channel with an alert message with channel listed (or some other sequence of response) It would be an intuitive for the user base.  could also be ~spam I guess, they would have to know the bot tag.
I won't bother with an FR if it is completely off the table. (edited)

The Librarian🔶 — 09/01/2021
Any reason it can't support multiple?
[12:36 PM]
Then people don't have to remember if it's @spam ~spam .spam etc

Sir_GoodEnough_🛠🔶 — 09/01/2021
I refuse to drop a "come up with something" FR... (edited)

1
[12:37 PM]
It probably can, but the @ may trigger discord, so that was the iffy one. (edited)

Sir_GoodEnough_🛠🔶 — 09/01/2021
I wonder if we made a trap channel...   These are likely bots spamming all the channels it can, right?  If there were a spam channel that had a topic for people not to post here, the spammer would anyway and we could watch that for activity. (edited)
[12:53 PM]
A post there would trigger the bot and sent it's alert?

The Librarian🔶 — 09/01/2021
That'd work
[12:55 PM]
Have the bot auto-mute anybody who posts there too

Sir_GoodEnough_🛠🔶 — 09/01/2021
maybe something like if you post anything here, you will be muted for an hour, and in that time we can ban them if they are bad. (edited)

1
[12:55 PM]
jinx

1

ⒿⓊⓈⓉⒾⓃ Ⓩ 🔶 — 09/01/2021
Honeypot channel, I like it

Sir_GoodEnough_🛠🔶 — 09/01/2021
I wonder how to get that to be the first one the bot sees?

The Librarian🔶 — 09/01/2021
First in the list, above welcome
[1:01 PM]
Make it hidden by anybody who has a role too?

Sir_GoodEnough_🛠🔶 — 09/01/2021
Then we also need a command to clear the channel, because if we clear it out manually we will be muted.
[1:14 PM]
like ~spamreset ...  maybe it clears that channel and shows the topic like .topic does as well where we say don't post here warning breathers off.
[1:15 PM]
This sounds like a sherlock bot, actually...

Blade2021🔶 — 09/01/2021
O.O
[3:17 PM]
Lets see... 

Yes we can make a command that does that.  Yes a FR needs to be submitted cause I have 4 different types of projects going at work alone which makes my brain scrambled as it is.  I wouldn't suggest a "honeypot" channel.  Its very... elusive and if I was a newcomer, I wouldn't know not to post there.
[3:17 PM]
Did I miss anything?

**Comment by Blade2021** [comment]
![image](https://user-images.githubusercontent.com/1572869/139558537-12555c50-2d30-47c3-96ff-ddf2e80fd6d4.png)
^ New infraction example

**Reply by Blade2021** [reply]
Also any ban events will trigger this automatically.

**Reply by Blade2021** [reply]
Warning Example:
![image](https://user-images.githubusercontent.com/1572869/139558554-32c6b118-02b1-4f43-aea5-78fc7b70dbb7.png)

**Comment by Blade2021** [comment]
![image](https://user-images.githubusercontent.com/1572869/139589710-3d9f7e6c-1d09-413f-9bed-d5b37ff2f336.png)
Setting a log channel will now automatically follow the Sherlock Updates channel for important updates and notices.

**Comment by Blade2021** [comment]
Today spam filtering was introduced.  As discord is feeling more and more of wraith from spammers, It is important that we continue to provide tools to detect, moderate, and log these types of attacks.

![image](https://user-images.githubusercontent.com/1572869/140236898-11d80227-eabf-4c3a-9043-6b3430ab589f.png)
![image](https://user-images.githubusercontent.com/1572869/140236920-e03dd13c-d8b5-45b2-a546-b412ceed949c.png)

**Comment by Blade2021** [comment]
Another form of Spam prevention that is in the works is Discord Server Invite protection.  Soon the ability to filter ALL discord invites will be at your fingertips.  Among this change, You will be able to whitelist GUILDS (Servers) that should be freely posted in your server.

There will be several moderation punishments that will come with these actions ranging from muting the individual for a certain amount of time, to automatic ban.

**Comment by Blade2021** [comment]
Another addition that is actively being developed is help descriptions for ALL commands.  This should make it easier for quick lookup of a command and its options/requirements.

![image](https://user-images.githubusercontent.com/1572869/140237306-20721233-d430-4415-a6d5-308451956491.png)

**Comment by Blade2021** [comment]
Well its been another busy morning!  Today it is released that going forward we will start protecting against rouge discord invites from users.
![image](https://user-images.githubusercontent.com/1572869/140616785-2d3da317-e9bd-44de-a36c-75460a556382.png)

**Comment by Blade2021** [comment]
![image](https://user-images.githubusercontent.com/1572869/141228923-78951148-7dc2-4ae2-91d6-b79ab8177e3a.png)
All Role related requests will be closely inspected by Sherlock to prevent any unauthorized members getting access to higher roles.

**Reply by Blade2021** [reply]
Example of Reply to user:
![image](https://user-images.githubusercontent.com/1572869/141229062-27598ed7-2414-468c-9404-a3ef2e4127d3.png)

**Comment by Blade2021** [comment]
I forgot to mention this previously but in order to make Sherlock cleaner for authorization of commands, we are rolling out two ways of providing access to commands.  

1. Custom moderator access
    The traditional method of creating access rights with mod roles will remain.  Each command will have an associated permission level that you will have to sum into your moderator access permission.

2. Discord Permissions
    Some commands will have discord permissions associated with them.  If a user/role has access to that permission, they will have access to any commands with said permission.

This will be specifically called out on any commands using the help command.  Example below.
![image](https://user-images.githubusercontent.com/1572869/141230181-22b7093d-dfe9-42c0-9186-2ee67b94357b.png)
GiveRole requires a mod permission index of 8 **OR** Any user/role with Manage Roles

**Comment by Blade2021** [comment]
Ban Updates!
![image](https://user-images.githubusercontent.com/1572869/142732346-78c33a82-3dd0-4be2-8188-25316b3c49a0.png)

After clicking the "unban" button
![image](https://user-images.githubusercontent.com/1572869/142732368-94ec9452-8f9a-4c46-be60-679b896db08b.png)

**Comment by Blade2021** [comment]
![image](https://user-images.githubusercontent.com/1572869/147880399-496cb412-a3ba-4b16-85e4-1314dac3f546.png)
has been returned to the previous log message:
![image](https://user-images.githubusercontent.com/1572869/147880410-e19abd34-6dcd-4d64-928c-217087d79f38.png)
